pkg/chunk/chunk_test.go

```diff
@@ -282,3 +302,76 @@ func benchmarkDecode(b *testing.B, batchSize int) {
 		}
 	}
 }
+
+func TestChunk_Slice(t *testing.T) {
+	now := model.Now()
+
+	// dummy chunk is created with time range now-1hour to now
+	chunkStartTime := now.Add(-time.Hour)
+
+	for _, tc := range []struct {
```

Comment:
All these tests have the `Start` and `End` exactly matching the timestamp of a sample in the chunk, because the whole logic is based on `now` and the samples interval is 15s. I would suggest add 1 test where the start and end doesn't exactly match the timestamp of a sample to have an higher confidence.